### ISSUE TITLE
deploy shoud slow faster if image pull is failing

### ISSUE THREAD

**Issue by matti** [kontena/kontena#2516]
```
Kontena::Workers::ImagePullWorker: image pull failed: denied: Permission denied for "latest" from request ...
...
Retrying still for 7 times.
```

this takes looong.

**Comment by SpComb** [comment]
Agreed. I assume there is some usecase for retrying the Docker image pull, but I don't know what it is. For the common case of permanent `Docker::Image.create!` errors like Not Found / Permission denied, where retrying will not help, the agent [`ImagePullWorker` retry logic](https://github.com/kontena/kontena/blob/v1.3.1/agent/lib/kontena/workers/image_pull_worker.rb#L40-L44) takes a looong time (several minutes?) to fail.

Either need to scrap the retry logic entirely, or if there is some known usecase of temporary errors where the agent retry logic is useful, fix it to only retry on those specific errors, not the common permanent ones?

**Comment by matti** [comment]
Yep. 300s to fail. Atleast permission problem should be safe to fail immediately?

**Comment by SpComb** [comment]
I noticed some odd `docker pull` behavior while the network was acting up, and SSL connections to index.docker.io were hanging. It was randomly either giving some network error... or something that looks suspisciously like a Docker API HTTP 404 error, which would result likely look like a `Docker::Error::NotFoundError` to the agent:

```
$ docker pull redis:latest
Pulling repository docker.io/library/redis
Tag latest not found in repository docker.io/library/redis
```

```
$ docker pull redis       
Using default tag: latest
Pulling repository docker.io/library/redis
Network timed out while trying to connect to https://index.docker.io/v1/repositories/library/redis/images. You may want to check your internet connection or if you are behind a proxy.
```

The agent seemed to pretty much hang on the image pull until I restarted it, even though I fixed the network and was able to `docker pull` again:

```
I, [2017-09-13T14:12:38.380987 #1]  INFO -- Kontena::Workers::ServicePodWorker: creating deptest-foo.redis-1
I, [2017-09-13T14:12:38.381063 #1]  INFO -- Kontena::ServicePods::Creator: creating service: deptest-foo.redis-1
I, [2017-09-13T14:12:38.381532 #1]  INFO -- Kontena::Workers::ImagePullWorker: pulling image: redis:latest
```
